blockmanager.go

```diff
@@ -3044,3 +3091,167 @@ func (l *lightHeaderCtx) RelativeAncestorCtx(
 		ancestorHeight, ancestor, l.store, l.headerList,
 	)
 }
+
+func (b *blockManager) sideLoadHeaders() {
+	// If sideloading is not enabled on this chain
+	// return quickly.
+	if !b.cfg.sideLoad.Enabled {
+		return
+	}
+
+	// If headers contained in the side load source are for a different chain network return
+	// immediately.
+	if b.sideLoadBlkHdrReader.HeadersChain() != b.cfg.ChainParams.Net {
+		log.Error("headers from side load file are of network %v "+
+			"and so incompatible with neutrino's current bitcoin network "+
+			"-- skipping side loading", b.sideLoadBlkHdrReader.HeadersChain())
+
+		return
+	}
+	log.Infof("Side loading headers from %v to %v", b.headerTip, b.sideLoadBlkHdrReader.EndHeight())
+
+	// Set headerTip to enable reader supply header, node needs
+	err := b.sideLoadBlkHdrReader.SetHeight(int64(b.headerTip))
```

Comment:
If our header tip is beyond the set of block headers, then we can just return early.

Reply:
Just like it is doing here. That is what the setHeight function does.

Reply:
I don't follow. If we have 100k blocks on start up, but the header source has only 10k headers, then we don't even need to look at the header source.

Reply:
Okay, but we need to first check right? how would we know in this case that the sideload source does not have the headers that we need? That is what this part does.